doc-content/drools-docs/src/main/asciidoc/DecisionEngine/cep-temporal-operators.adoc

```diff
@@ -0,0 +1,457 @@
+[id='cep-temporal-operators_{context}']
+
+= Temporal operators for events
+
+In stream mode, the {DECISION_ENGINE} supports the following temporal operators for events that are inserted into the working memory of the {DECISION_ENGINE}. You can use these operators to define the temporal reasoning behavior of the events that you declare in your Java class or DRL rule file. Temporal operators are not supported when the {DECISION_ENGINE} is running in cloud mode.
+
+* `after`
+* `before`
+* `coincides`
+* `during`
+* `includes`
+* `finishes`
+* `finishes by`
+* `meets`
+* `met by`
+* `overlaps`
+* `overlapped by`
+* `starts`
+* `started by`
+
+after::
++
+--
+This operator defines an amount of time after which the current event can follow the correlated event, or a delimiting time range during which the current event can follow the correlated event.
+
+For example, the following pattern matches if `$eventA` starts between 3 minutes and 30 seconds and 4 minutes after `$eventB` finishes. If `$eventA` starts earlier than 3 minutes and 30 seconds after `$eventB` finishes, or later than 4 minutes after `$eventB` finishes, then the pattern is not matched.
+
+[source]
+----
+$eventA : EventA(this after[3m30s, 4m] $eventB)
+----
+
+You can also express this operator in the following way:
+
+[source]
+----
+3m30s <= $eventA.startTimestamp - $eventB.endTimeStamp <= 4m
+----
+
+The `after` operator supports up to two parameter values:
+
+* If two values are defined, the interval starts on the first value (3 minutes and 30 seconds in the example) and ends on the second value (4 minutes in the example).
+* If only one value is defined, the interval starts on the provided value and runs indefinitely with no end time.
+* If no value is defined, the interval starts at 1 millisecond and runs indefinitely with no end time.
+
+The `after` operator also supports negative time ranges:
+
+[source]
+----
+$eventA : EventA(this after[-3m30s, -2m] $eventB)
+----
+
+If the first value is greater than the second value, the {DECISION_ENGINE} automatically reverses them. For example, the following two patterns are interpreted by the {DECISION_ENGINE} in the same way:
+
+[source]
+----
+$eventA : EventA(this after[-3m30s, -2m] $eventB)
+$eventA : EventA(this after[-2m, -3m30s] $eventB)
+----
+--
+
+before::
+This operator defines an amount of time after which the current event can precede the correlated event, or a delimiting time range during which the current event can precede the correlated event.
```

Comment:
No, you're right, it's confusing. I was trying to say the current event comes before correlated event _after_ x amount of time passes. But that's confusing. I've clarified the wording like what you have. I also turned the description and example around so that it talks about the current event happening before the correlated one, instead of the correlated one happening after the current one. They mean the same thing, and I did the second one to begin with because that's the pattern I've used in the other descriptions (where I just swap the current and correlated events in the explanations), but in this case, it's confusing to talk about things happening "after" another in a section about "before."